var/spack/repos/builtin/packages/py-cupy/package.py

```diff
@@ -47,7 +54,10 @@ class PyCupy(PythonPackage, CudaPackage, ROCmPackage):
     # Based on https://github.com/cupy/cupy/releases
     depends_on("cuda@:11.9", when="@:11 +cuda")
     depends_on("cuda@:12.1", when="@12:12.1.0 +cuda")
-    depends_on("cuda@:12.4", when="@13: +cuda")
+    depends_on("cuda@:12.1", when="@13.0 +cuda")
+    depends_on("cuda@:12.4", when="@13.1.0:13.2.0 +cuda")
+    depends_on("cuda@:12.6", when="@13.3.0 +cuda")
+    depends_on("cuda@:12.8", when="@13.4.0: +cuda")
```

Comment:
Same as the other comment, possibly useless, but helpful in case of patch releases.
```suggestion
    depends_on("cuda@:12.1", when="@13.0 +cuda")
    depends_on("cuda@:12.4", when="@13.1:13.2 +cuda")
    depends_on("cuda@:12.6", when="@13.3 +cuda")
    depends_on("cuda@:12.8", when="@13.4: +cuda")
```